src/main/java/org/openpnp/machine/reference/feeder/Bank.java

```diff
@@ -0,0 +1,271 @@
+import com.google.zxing.common.HybridBinarizer;
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+
+import org.openpnp.model.Configuration;
+import org.openpnp.model.Location;
+import org.openpnp.model.LengthUnit;
+import org.simpleframework.xml.*;
+import org.simpleframework.xml.core.Commit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.openpnp.gui.MainFrame;
+import org.openpnp.gui.components.*;
+import org.openpnp.model.Configuration;
+
+
+import org.openpnp.gui.support.PropertySheetWizardAdapter;
+import org.openpnp.gui.support.Wizard;
+import org.openpnp.machine.reference.ReferenceFeeder;
+import org.openpnp.machine.reference.ReferenceMachine;
+import org.openpnp.spi.*;
+import org.openpnp.machine.reference.feeder.wizards.ReferenceFeederConfigurationWizard;
+
+import org.openpnp.spi.base.SimplePropertySheetHolder;
+
+
+
+
+
+
+public class Bank extends ReferenceFeeder {
+    private final static Logger logger = LoggerFactory.getLogger(Bank.class);
+    private final double cam_x = 10.0; // mm
+    private final double cam_y = 10.0; // mm
+
+    @Element(required=true)
+    private Location offset=new Location(LengthUnit.Millimeters);
+
+    @ElementList(required = false)
+    protected IdentifiableList<Feeder> feeders = new IdentifiableList<>();
+
+    private ReferenceMachine machine;
+
+    @Override
+    public Location getPickLocation() throws Exception {
+        return location;
+    }
+
+
+    //@SuppressWarnings("unused")
+    //@Commit
+    private void commit() {
+	boolean ena=enabled;
+	//if(!ena) mapLoc(false);
+	setEnabled(false);
+	setEnabled(ena);
+    }
+
+  public Bank() {
+        Configuration.get().addListener(new ConfigurationListener.Adapter() {
+            @Override
+            public void configurationLoaded(Configuration configuration) throws Exception {
+		commit();
+            }
+        });
+    }
+
+
+    @Override
+    public void setEnabled(boolean enabled) {
+	ReferenceMachine machine = (ReferenceMachine)Configuration.get().getMachine();
+	String id=getId() + ":"; int i=-1;
+	if(this.enabled!=enabled) {
+		//mapLoc(this.enabled=enabled); 	// this stage
+	  	for(Feeder fdr:feeders) 	// deeper stages
+		if(fdr.getClass()==this.getClass())  // this class
+		if(this.enabled) ((Bank)fdr).commit(); else {
+			enabled=((Bank)fdr).enabled;
+			fdr.setEnabled(false);
+			((Bank)fdr).enabled=enabled;
+		}
+	}
+	if(!enabled) {
+	  for(Feeder fdr:feeders) {
+		machine.removeFeeder(fdr);
+	  }
+	  for(Feeder fdr:machine.getFeeders()) 
+		if(fdr.getName().startsWith(id))
+		machine.removeFeeder(fdr);
+	}
+
+	if( enabled) 
+	for(Feeder fdr:feeders) {
+			if(machine.getFeeder(fdr.getId())!=fdr) 
+			try { machine.addFeeder(fdr); } catch(Exception e) {;}
+	}
+    }
+
+
+    @Override
+    public void feed(Nozzle nozzle) throws Exception {
+	if(getId().length()==0) { // hack
+		setEnabled(true);
+		feeders.clear();
+		for(Board b:Configuration.get().getBoards()) 
```

Comment:
Loops and conditionals must use braces. Please see https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style and clean up the code.